### ISSUE TITLE
whoisd crashes when listening on IPv6

### ISSUE THREAD

**Issue by toreanderson** [SpriteLink/NIPAP#718]
When the config file contains `listen = ::` (or `listen = [::]`), nipap-whoisd crashes at startup:

```
$ nipap-whoisd --config /etc/nipap/whoisd.conf
Traceback (most recent call last):
  File "/usr/sbin/nipap-whoisd", line 180, in <module>
    except socket.error, exc:
NameError: name 'socket' is not defined
```

If I change the config file to say `listen = 0.0.0.0`, it starts fine. The full whoisd.conf is found below:

```
[whoisd]
user = nobody
group = nogroup
listen = ::
port = 43
pid_file = /var/run/nipap-whoisd/whoisd.pid
nipapd_host = localhost
nipapd_port = 1337
nipapd_username = whoisd-user
nipapd_password = foo
```

**Comment by plajjan** [comment]
So there are multiple bugs here. One is that we are trying to catch a socket.error exception (https://github.com/SpriteLink/NIPAP/blob/master/whoisd/nipap-whoisd#L180) and thus are referencing the socket module but haven't imported it.

Second thing is that the SocketServer that ships with Python 2.7 simply doesn't support IPv6. It is hard-coded to use IPv4 :( see http://bugs.python.org/issue20215

**Comment by plajjan** [comment]
Socket import fixed in 653f09aa249021d778d89a4aa4567593b2925f50

**Comment by plajjan** [comment]
Hmm, I think I can monkey patch this thing to support IPv6 but it might mean one has to supply an IPv4/v6 mapped address for listening to an IPv4-address....
